File: XMLCheck/Handler/CheckRulesXML05/Xml5RuleRunner.cs
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML05
{
    public static class Xml5RuleRunner
    {
        private static readonly List<IXml5Rule> _rules = new()
        {

        };
        public static List<ErrorDetails> Run(List<Xml5Model>? models,HoSoContext context)
        {
            var errors = new List<ErrorDetails>();

            if (models == null || models.Count == 0)
                return errors;

            foreach (var model in models)
            {
                foreach (var rule in _rules)
                {
                    var err = rule.Check(model,context);
                    if (err != null)
                        errors.Add(err);
                }
            }

            return errors;
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml5/xml5RuleRunner.ts
import { Xml5Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';
import { IXml5Rule } from './xml5RuleBase';

export class Xml5RuleRunner {
  private static readonly rules: IXml5Rule[] = [];

  public static run(models: Xml5Model[], context: HoSoContext): ErrorDetails[] {
    const errors: ErrorDetails[] = [];
    if (!models || models.length === 0) return errors;

    for (const model of models) {
      for (const rule of this.rules) {
        const err = rule.check(model, context);
        if (err) {
          errors.push(err);
        }
      }
    }

    return errors;
  }
}
